### PR TITLE
UWP improvements

### PR DESCRIPTION
**Pull request by f1nzer**
ref #354 #366
- moved all logging to single method
- improved CustomOverlay logic
- fixed demo ui
- fixed single scan after continious scanning (all scans were continious)
- added log message to notify user about NavigationCacheMode if it's not enabled.

Also added default software/hardware back button handler. Typically application already has handlers for this button and always it's defined on startup, so we can add new handler which will be called after default one (if exists) according to IsHandled value.
Previously I've tried to get GoBackRequested event from SystemNavigationManager via reflection and check listeners (in order to subscribe only in case if there are no listeners) but had no success to get that field (always FieldInfo was null).

For now something wrong with my uwp device so would be great if anyone can test back button now (without default handler). @huangjinshe

### CODE REVIEW

File: Source/ZXing.Net.Mobile.WindowsUniversal/MobileBarcodeScanner.cs [hunk 1]
```diff
@@ -3,53 +3,72 @@
...
+        {
+            get
+            {
+                var currentFrame = RootFrame ??
+                                   Window.Current.Content as Frame ??
+                                   ((FrameworkElement) Window.Current.Content).GetFirstChildOfType<Frame>();
+
+                var currentPage = currentFrame.Content as Page;
+                if (currentPage != null && currentPage.NavigationCacheMode == NavigationCacheMode.Disabled)
+                    Log("WARNING: you may need to set {0}=\"Enabled\" for your Page to solve problems with updating UI after scanning is completed",
```
Comment (F1nZeR) [inline Source/ZXing.Net.Mobile.WindowsUniversal/MobileBarcodeScanner.cs]: Logging now implemented via Debug.WriteLine so it's will be not compiled on release. We might use Console.WriteLine (is it works here?) or mby (especially for NavigationCacheMode and other critical moments) other mechanism.

**Comment by huangjinshe** [comment]
Yesterday, I take time and add GoBackRequested event with NavigationCacheMode as you mention, It's work fine now. did you change another logic now? or just  improve?  What you want me to test?

**Comment by F1nZeR** [comment]
For now back press shouldn't close app even if you have no GoBackRequested implementation.
Also it's should work properly if you've implemented it.

**Comment by huangjinshe** [comment]
I think you're right, we should always use GoBackRequested  event. for real back button.

I think you just need to make sure it will no any problem no matter developers implement GoBackRequested  or not. :-)

**Comment by F1nZeR** [comment]
So this PR contains this logic, but I have no phone to test it :smile:

**Comment by huangjinshe** [comment]
I donwload your code and try to test.
1.Don't you change these based on my code?  where is the cancel button logic which I put it?
2.I can't compile your source code, visual studio always output:
```
Conflicting values for resource 'Files/ZXingNetMobile/ZXingScannerControl.xbf'	App1.UWP	Y:\Download\sample of barcode problem3\App1\App1.UWP\MakePRI		
MakePri : error 0x80073b0f: Processing Resources failed with error:duplicate entry.
```

**Comment by Redth** [comment]
I don't think we should put a cancel button in. This is something users can add to their own custom overlay if they want. On Android we have no cancel button since there's a hardware back button. Same with UWP on mobile.

**Comment by F1nZeR** [comment]
@Redth but what the reason to have properties `FlashButtonText` and `CancelButtonText` in `MobileBarcodeScannerBase` implementation?

**Comment by huangjinshe** [comment]
I just think in the future maybe there is no always a hardware back button on the phone.  Also maybe some user just like click button in the screen, not press the real back button on the phone.

**Pull request closed without merging**